python/lsst/meas/extensions/shapeHSM/_hsm_shape.py

```diff
@@ -273,7 +273,9 @@ def measure(self, record, exposure):
                 guessCentroid._p,  # guess_centroid
                 hsmparams._hsmp,  # hsmparams
             )
-        except galsim.hsm.GalSimHSMError as error:
+        # GalSim does not raise custom pybind errors as of version 2.5,
+        # resulting in all GalSim C++ errors being RuntimeErrors.
+        except (galsim.hsm.GalSimHSMError, RuntimeError) as error:
```

Comment:
Is there any concern that these RuntimeError's raised by GalSim might be more serious ones that we should further investigate, or can we assume that they're all just internal algorithmic problems and ignore them (which is what this is effectively doing)?

Reply:
The errors were mostly being thrown on objects that were flagged but not ignored due to the bug above, so I'm not too worried about that. They are unfortunately not granular so if you wanted to add more specific flags you'd need to parse the error message.

Note that the comment above mentions DM-42047, which will change the line above to call GalSim's Python interface. But I see that [that interface](https://github.com/GalSim-developers/GalSim/blob/releases/2.5/galsim/hsm.py#L700) just catches `RuntimeError` and re-raises as `GalSimHSMError`, which will end up in the same result.

Reply:
I also agree that we can assume these are internal GalSim issues that are expected for some objects and don't need further investigation from us. Additionally, parsing the error message and creating more flags doesn't seem future-proof to me.